src/main/java/org/adrian/com/spring/models/BinarySearchRequest.java

```diff
@@ -0,0 +1,24 @@
+package org.adrian.com.spring.models;
+
+import java.util.Arrays;
+
+public record BinarySearchRequest(int[] array, int expected) {
+    @Override
+    public boolean equals(Object obj) {
```

Comment:
Generuj to lombokiem, i tak widzę po metodzi hashCode że generowałeś przez IDE (intelliJ zawsze używa magic number 31 do skrótu hash)